timbuctoo-instancev4/src/main/java/nl/knaw/huygens/timbuctoo/search/description/indexes/PersonIndexDescription.java

```diff
@@ -17,14 +26,60 @@
     "modified"
   };
 
+  private final List<String> types;
+  private final PropertyParserFactory propertyParserFactory;
+  private final HashMap<String, PropertyParser> parsers;
+
+  public PersonIndexDescription(List<String> types) {
+    this.types = types;
+    this.propertyParserFactory = new PropertyParserFactory();
+    this.parsers = Maps.newHashMap();
+    parsers.put("names", propertyParserFactory.getParser(PersonNames.class));
+    parsers.put("deathDate", propertyParserFactory.getParser(Datable.class));
+    parsers.put("birthDate", propertyParserFactory.getParser(Datable.class));
+    parsers.put("modified", propertyParserFactory.getParser(Change.class));
+  }
+
   @Override
-  public List<String> getSortIndexPropertyNames(List<String> vertexTypes) {
-    List<String> fieldList = new ArrayList<>();
-    for (String type : vertexTypes) {
+  public Set<String> getSortIndexPropertyNames() {
+    Set<String> fieldList = new HashSet<>();
+    for (String type : types) {
       for (String field : SORT_FIELDS) {
-        fieldList.add(String.format("%s_%s_%s", type, field, SORT_POSTFIX));
+        fieldList.add(getSortPropertyName(type, field));
       }
     }
     return fieldList;
   }
+
+  @Override
+  public void addIndexes(Vertex vertex) {
+    for (String type : types) {
```

Comment:
Hoe kunnen we zorgen dat dit niet meer magische tokens zijn ("names" en "deathDate") of gedupliceerde constructielogica ("wwperson_names")